File: 17950.py
import sys

MOD = 1000000007
H, x = map(int, sys.stdin.readline().split())

sum = 0; snowball = 1
for i in range(H):
    c = int(sys.stdin.readline())

    snowball *= x
    snowball %= MOD
    sum += snowball * c
    sum %= MOD

print(sum)

File: 17950.cpp
#include <iostream>
using namespace std;

const int MOD = 1000000007;

int main() {
	int H, x; cin >> H >> x;

	long long sum, snowball;
	sum = 0; snowball = 1;
	for (int i = 0; i < H; i++) {
		int c; cin >> c;

		snowball *= x;
		snowball %= MOD;
		sum += snowball * c;
		sum %= MOD;
	}

	cout << sum % 1000000007 << endl;
}
